xtask/codegen/gritql.ungram

```diff
@@ -0,0 +1,554 @@
+// GritQL Un-Grammar.
+//
+// This grammar specifies the structure of Rust's concrete syntax tree.
+// It does not specify parsing rules (ambiguities, precedence, etc are out of scope).
+// Tokens are processed -- contextual keywords are recognised, compound operators glued.
+//
+// Legend:
+//
+//   //          				-- comment
+//   Name =      				-- non-terminal definition
+//   'ident'     				-- token (terminal)
+//   A B         				-- sequence
+//   A | B       				-- alternation
+//   A*          				-- zero or more repetition
+//   (A (',' A)* ','?)	        -- repetition of node A separated by ',' and allowing a trailing comma
+//   (A (',' A)*)	            -- repetition of node A separated by ',' without a trailing comma
+//   A?          				-- zero or one repetition
+//   (A)         				-- same as A
+//   label:A     				-- suggested name for field of AST node
+
+// NOTES
+//
+// - SyntaxNode, SyntaxToken and SyntaxElement will be stripped from the codegen
+// - Bogus nodes are special nodes used to keep track of broken code; they are
+//   not part of the grammar but they will appear inside the green tree
+//
+
+
+///////////////
+// Bogus NODES
+///////////////
+// SyntaxElement is a generic data structure that is meant to track nodes and tokens
+// in cases where we care about both types
+//
+// As Bogus* node will need to yield both tokens and nodes without discrimination,
+// and their children will need to yield nodes and tokens as well.
+// For this reason, SyntaxElement = SyntaxElement
+SyntaxElement = SyntaxElement
+
+GritBogus = SyntaxElement*
+GritBogusContainer = SyntaxElement*
+GritBogusDefinition = SyntaxElement*
+GritBogusPattern = SyntaxElement*
+GritBogusLiteral = SyntaxElement*
+GritBogusNamedArg = SyntaxElement*
+GritBogusPredicate = SyntaxElement*
+
+GritRoot =
+    bom: 'UNICODE_BOM'?
+    version: GritVersion?
+    language: GritLanguageDeclaration?
+    definitions: GritDefinitionList?
+    pattern: AnyGritPattern?
+    definitions_continued: GritDefinitionList?
+    eof: 'EOF'
+
+GritSequential = 'sequential' '{' sequential: GritPatternList '}'
```

Comment:
I **highly** suggest adding examples to most of the node. This has helped with the reviews, and it also helped to give proper context to fellow developers. Check `js.ungram` and `css.ungram` for examples. 

The example should be a comment that shows a piece of the grammar, and **highlight** the piece of the grammar that the node covers.